generated/nixnet/nixnet_service.cpp

```diff
@@ -359,8 +361,9 @@ namespace nixnet_grpc {
     }
     try {
       auto database_ref_grpc_session = request->database_ref();
-      nxDatabaseRef_t database_ref = session_repository_->access_session(database_ref_grpc_session.id(), database_ref_grpc_session.name());
+      nxDatabaseRef_t database_ref = nx_database_ref_t_resource_repository_->access_session(database_ref_grpc_session.id(), database_ref_grpc_session.name());
       u32 close_all_refs = request->close_all_refs();
```

Comment:
If `close_all_refs` is true, this library call sets reference counter for this db reference to 0, but we will have dangling entries in `nx_database_ref_t_resource_repository_`. Does it make sense to custom implement `DbCloseDatabase` and when `close_all_refs` is true, iterate over `nx_database_ref_t_resource_repository_` find all entries which have value of `database_ref ` and purge them? If we don't do this, `nx_database_ref_t_resource_repository_` could potentially keep growing in size.

Reply:
I think we should leave it as-is. We can consider adding that as a feature in the future. I would expect `close_all_refs` to be an infrequently used feature and the consequence is minor. And there are 2 "workarounds" by using grpc-device-style APIs: use named sessions so there's only one grpc-device session or call reset_server to reset everything.

Is `close_all_refs` used in xnet examples? Or recommended anywhere? My understanding is that those "force close" APIs are always discouraged because they circumvent ref counting. But maybe there are some APIs where they are more accepted?

Reply:
`close_all_refs` is used by all `nxDbCloseDatabase` in examples.

There is this note in documentation about when `close_all_refs` should be used
> Another option is that only one thread executes nxdbCloseDatabase once, using CloseAllRefs = 1, which closes access for all other threads. This may be convenient when, for example, the main program needs to stop all running threads and be sure the database is closed properly, even if some threads could not execute nxdbCloseDatabase.

But I agree that consequence does seem minor here, so we could leave that for future.